app/middleware/earthdata-login-oauth-authorizer.ts

```diff
@@ -43,6 +43,12 @@ const oauthOptions = {
  * @param _next - The next function in the middleware chain
  */
 async function handleCodeValidation(oauth2: OAuthClient, req, res, _next): Promise<void> {
+  const { state } = req.signedCookies;
+
+  if (state !== req.query.state) {
+    throw new RequestValidationError();
```

Comment:
I think we want to put in a detailed error message in the RequestValidationError. e.g. 
`throw new RequestValidationError('OAuth state mismatch');`